### PR TITLE
Adds the ability to (de)construct the pool ladder

### PR DESCRIPTION
**Pull request by Bokkiewokkie**
## About The Pull Request
Originally made this to fix the ladder being unanchored, but figured adding (de)construction steps to it could be helpful too.
Deconstruction is done with a wrench.
Right now you can only build them from outside the pool when standing north of it (because ladders don't turn).
Also changed the pool part names to be lowercase, since they're not proper names, and the description of the pool ladder to be less OOC (mentioning clicking on it).

## Why It's Good For The Game
Ladders are more useful when secured, and when you can rebuild them!

## Testing Photographs and Procedure

https://github.com/BeeStation/BeeStation-Hornet/assets/43698041/7da7f608-af55-4b84-8e56-1adc36dad04f

## Changelog
:cl:
add: Made it possible to disassemble pool ladders with wrenches and possible to build them with rods.
fix: Fixed pool ladders being unanchored.
/:cl:

### CODE REVIEW

File: code/modules/pool/pool.dm [hunk 1]
```diff
@@ -96,6 +96,16 @@ Place a pool filter somewhere in the pool if you want people to be able to modif
 	if(do_after(user, 4 SECONDS, target = dropping))
 		splash(dropping)
 
+/turf/open/indestructible/sound/pool/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	if(istype(W, /obj/item/stack/rods) && user.loc == get_step(src, NORTH) && !istype(user.loc, /turf/open/indestructible/sound/pool/))
```
Comment (PowerfulBacon) [inline code/modules/pool/pool.dm]: This should check to ensure that multiple pool ladders aren't being placed on the same tile. It also seems pretty restrictive to only be placed it in the solo tile above it. I think that it should simply check if the tile to the north step is not a pool tile, if so then place the ladder. I don't think it matters if someone starts building a ladder while in a pool, but if you wan't to block that then you can keep the user.loc check.

File: code/modules/pool/pool.dm [hunk 2]
```diff
@@ -96,6 +96,16 @@ Place a pool filter somewhere in the pool if you want people to be able to modif
...
+		var/obj/item/stack/rods/R = W //^ Only build pool ladders from the shore (north, because the ladder can't rotate)
+		if(R.tool_use_check(user,10))
+			to_chat(user, "<span class ='notice'>You start installing a pool ladder.</span>")
+			if(do_after(user, 5 SECONDS, target=src))
+				R.use(10)
+				new /obj/structure/pool_ladder(src)
+				return TRUE
```
Comment (PowerfulBacon) [inline code/modules/pool/pool.dm]: This would look a lot cleaner with some early returns

**Review by PowerfulBacon — COMMENTED** [review]
Needs some simple testing evidence

File: code/modules/pool/pool.dm (hunk 1 above)
Comment (Bokkiewokkie) [reply code/modules/pool/pool.dm]: Yeah I'll cut some of that

**Comment by Bokkiewokkie** [comment]
> Needs some simple testing evidence

I did test it a few times, but I can record a video

**Review by PestoVerde322 — APPROVED** [review]
Code seems good to go, Approved

File: code/modules/pool/pool.dm [hunk 3]
```diff
@@ -92,10 +92,26 @@ Place a pool filter somewhere in the pool if you want people to be able to modif
...
+	if(..())
+		return
+	if(!istype(W, /obj/item/stack/rods))
+		return
+	if(locate(/obj/structure/pool_ladder) in src)
+		balloon_alert(user, "You try to make a pool ladder, but there is aleady one here!")
+		return
+	if(!W.tool_use_check(user,10))
+		return
+	if(!istype(get_step(src, NORTH), /turf/open/indestructible/sound/pool/)) //Ladders only face up, and no stacking!
```
Comment (PowerfulBacon) [inline code/modules/pool/pool.dm]:
```suggestion
	if(!istype(get_step(src, NORTH), /turf/open/indestructible/sound/pool)) //Ladders only face up, and no stacking!
```

**Review by PowerfulBacon — APPROVED** [review]

**Comment by PowerfulBacon** [comment]
Thanks for bringing this upstream

**Comment by Bokkiewokkie** [comment]
> Thanks for bringing this upstream

Always happy to

**Pull request merged**